Failing assertion with CoreALT routing

CALT routing might fail throwing the following error when run-time assertions are enabled by the `-ea` flag . The problem seems to originate from an attempt of checking whether a virtual node is a core node.

```
Caused by: java.lang.AssertionError: node 11067 is invalid. Not in [0,11066)
	at com.graphhopper.storage.CHGraphImpl.checkNodeId(CHGraphImpl.java:120) ~[graphhopper-core-v0.13.15.jar:?]
	at com.graphhopper.storage.CHGraphImpl.getLevel(CHGraphImpl.java:115) ~[graphhopper-core-v0.13.15.jar:?]
	at org.heigit.ors.routing.graphhopper.extensions.core.AbstractCoreRoutingAlgorithm.isCoreNode(AbstractCoreRoutingAlgorithm.java:203) ~[classes/:?]
	at org.heigit.ors.routing.graphhopper.extensions.core.CoreALT.fillEdgesFrom(CoreALT.java:131) ~[classes/:?]
...
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -60,6 +60,7 @@ RELEASING:
 - `ORS_APP_CONF` environment variable ([#1017](https://github.com/GIScience/openrouteservice/issues/1017))
 ### Fixed
 - Errors in travel speed explanation
+- Failing assertion with CALT routing ([#1047](https://github.com/GIScience/openrouteservice/issues/1047))
 
 ## [6.6.1] - 2021-07-05
 ### Fixed

--- a/openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/core/AbstractCoreRoutingAlgorithm.java
+++ b/openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/core/AbstractCoreRoutingAlgorithm.java
@@ -60,7 +60,8 @@ public abstract class AbstractCoreRoutingAlgorithm extends AbstractRoutingAlgori
         int size = Math.min(2000, Math.max(200, graph.getNodes() / 10));
         initCollections(size);
 
-        chGraph = (CHGraph) ((QueryGraph) graph).getMainGraph();
+        qGraph = (QueryGraph) graph;
+        chGraph = (CHGraph) qGraph.getMainGraph();
         coreNodeLevel = chGraph.getNodes() + 1;
         turnRestrictedNodeLevel = coreNodeLevel + 1;
     }
@@ -68,6 +69,7 @@ public abstract class AbstractCoreRoutingAlgorithm extends AbstractRoutingAlgori
     protected abstract void initCollections(int size);
     protected PathBidirRef bestPath;
 
+    QueryGraph qGraph;
     CHGraph chGraph;
     protected final int coreNodeLevel;
     protected final int turnRestrictedNodeLevel;
@@ -200,7 +202,7 @@ public abstract class AbstractCoreRoutingAlgorithm extends AbstractRoutingAlgori
     }
 
     boolean isCoreNode(int node) {
-        return chGraph.getLevel(node) >= coreNodeLevel;
+        return !qGraph.isVirtualNode(node) && chGraph.getLevel(node) >= coreNodeLevel;
     }
 
     boolean isTurnRestrictedNode(int node) {

